### ISSUE TITLE
Lint Option: File Access should be completed using SQL

### ISSUE THREAD

**Issue by priceaj** [halcyon-tech/vscode-rpgle#10]
SETLL, CHAIN READ etc should be flagged and the SQL equilivalents suggested

**Comment by worksofliam** [comment]
LOVE this one. 

@priceaj Would you perhaps be willing to share the SQL on this issue?

e.g. 

* `CHAIN` becomes `SELECT...`

**Comment by bmorris1** [comment]
Usually it's not a great idea to just replace each RPG statement with a matching SQL statement. That can lead to terrible performance. But it would be super complex (if it's even possible to do programmatically) to suggest how to do the all the I/O for some file with one or more SQL statements.

**Comment by priceaj** [comment]
Hi @bmorris1 

I'm quite surprised to hear someone from the RPG development team at IBM mentioning performance issues between SQL and RLA, especially when the "modernization" content I have read coming from IBM seems to be pushing people towards embedded SQL and away from RLA ("Modernizing IBM i Applications from the Database up to the User Interface and Everything in Between", springs to mind!) 

Is it the blind programmatic conversion that you would be concerned about or the concept of moving completely away from RLA?

**Comment by bmorris1** [comment]
Yes, it's the blind conversions of each opcode to an SQL equivalent. I was disagreeing with the idea that the linter would suggested an equivalent SQL operation for each RPG operation.

If people intelligently change what the RPG logic is doing to do the same thing in SQL in a set-based way, then the SQL performance might be much much better than the RPG could ever be.

**Comment by priceaj** [comment]
@bmorris1 phew! Thanks for confirming!!

**Comment by worksofliam** [comment]
@priceaj With all that being said, how do you see this working now?

**Comment by priceaj** [comment]
So we are still making the linter Unopinionated Right? (As in this won't be enabled by default)

Just have it highlight when the relevant RLA op codes are used, you don't have to provide a quick fix for everything. Most linters don't they will just say, this is wrong, fix it.

It's the choice of the user to enable it!

**Comment by worksofliam** [comment]
@priceaj Right! There is a default configuration that is created when people enable it: https://github.com/halcyon-tech/vscode-rpgle/blob/main/src/models/default.js

Maybe this is similar to #11?
